mavisp/methods.py

```diff
@@ -178,33 +177,70 @@ class RosettaDDGPredictionStability(Method):
+                                        critical=[MAVISpCriticalError(this_error)])
 
-        try:
-            mutation_data = pd.read_csv(os.path.join(dir_path, rosetta_file))
-        except Exception as e:
-            this_error = f"Exception {type(e).__name__} occurred when parsing the Rosetta csv file. Arguments:{e.args}"
-            raise MAVISpMultipleError(warning=warnings, 
-                                      critical=[MAVISpCriticalError(this_error)])
+            mutation_data = mutation_data[mutation_data['state'] == 'ddg']
+            mutation_data = mutation_data.set_index('mutation_label')
+            mutation_data = mutation_data[['total_score']]
+            mutation_data = mutation_data.rename(columns={'total_score':f'{self.type} ({self.version}, {self.unit})'})
 
-        mutation_data = mutation_data[mutation_data['state'] == 'ddg']
-        mutation_data = mutation_data.set_index('mutation_label')
-        mutation_data = mutation_data[['total_score']]
-        mutation_data = mutation_data.rename(columns={'total_score':f'{self.type} ({self.version}, {self.unit})'})
+            self.data = mutation_data
+
+            if len(warnings) > 0:
+                raise MAVISpMultipleError(warning=warnings, 
+                                        critical=[])
+        else:
+            csv_files = []
+            rosetta_folder = os.listdir(dir_path)
+            # In order to browse the rosetta file, and access the cl folders
+            for folder in rosetta_folder:
+                ddg_file = os.listdir(os.path.join(dir_path, folder))
+                if len(ddg_file)!=1:
+                    this_error = f"multiples files found in {dir_path}; only one expected"
+                    raise MAVISpMultipleError(warning=warnings,
+                                              critical=[MAVISpCriticalError(this_error)])
+            # Gather all csv files
+                csv_files.extend([os.path.join(dir_path, folder, file) for file in ddg_file if file.endswith(".csv")])
+            try:
+                mutation_data = pd.DataFrame()
+                for file in csv_files:
+                    tmp = pd.read_csv(file)
+                    mutation_data = pd.concat([mutation_data,tmp])
+                    # Allow to merge the data from the different cl folders
+                mutation_data = mutation_data.groupby(["mutation_label","state"])[mutation_data.columns[6:]].agg('mean')
+                # Sort the data by mutation_label and state, and calculate the mean of the different ddg values
+                mutation_data.sort_values(by=['mutation_label','state'], inplace=True)
+                mutation_data = mutation_data.reset_index()
+                mutation_data = mutation_data[['total_score', 'mutation_label']]
+                self.data = mutation_data.rename(columns={'total_score':f'{self.type} ({self.version}, {self.unit})'})
+                self.data = self.data.set_index('mutation_label')
```

Comment:
* in these csv files we are only interested in the lines that have `ddg` in the `state` column; the other ones aren't free energy differences and need to be discarded, plus we are only interested in the `total_score` and `mutation_label` columns. I would do just do filtering of rows and columns at first so we discard immediately what we don't need and make the processing easier/faster for the next steps.

* before doing the averaging, can we check that the all the csv have the same `mutation_label` column (i.e. they contain the same dataset(s))?

* If you do the filtering as per previous point, you can simplify the data merging a bit